featurebyte/api/item_table.py

```diff
@@ -299,21 +312,28 @@ def get_view(
             timestamp_timezone_offset_column_name=timestamp_timezone_offset_column,
         )
 
-    @root_validator(pre=True)
-    @classmethod
-    def _set_default_feature_job_setting(cls, values: dict[str, Any]) -> dict[str, Any]:
-        if "event_table_id" in values:
-            event_table_id = values["event_table_id"]
-            try:
-                default_feature_job_setting = EventTable.get_by_id(
-                    event_table_id
-                ).default_feature_job_setting
-            except RecordRetrievalException:
-                # Typically this shouldn't happen since event_table_id should be available if the
-                # ItemTable was instantiated correctly. Currently, this occurs only in tests.
-                return values
-            values["default_feature_job_setting"] = default_feature_job_setting
-        return values
+    @property
+    @cachedmethod(cache=operator.attrgetter("_cache"), key=get_default_job_setting_cache_key)
+    def default_feature_job_setting(self) -> Optional[FeatureJobSetting]:
+        """
+        Returns the default feature job setting for the table.
```

Comment:
@jevonyeoh Due to the refactoring the docstring will be extracted from here instead